File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.cpp
class Solution {
public:
    string reversePrefix(string word, char ch) {
        int pos, nextPos;
        bool flag = false;
        for (int i = 0; i < word.size(); i++)
        {
            if(word[i] == ch)
            {
                flag = true;
                pos = i;
                nextPos = pos + 1;
                break;
            }
        }
        if(flag){
            string updatedWord = "";
            for (int i = pos; i >= 0; i--)
                updatedWord+=word[i];
            if (pos != word.size() - 1)
            {
                for (int i = nextPos; i < word.size(); i++)
                    updatedWord+=word[i];
            }
            return updatedWord;
        }
        else
        return word;
    }
};

File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.java
class Solution {
    public String reversePrefix(String word, char ch) {
        int pos = 0, nextPos = 1;
        boolean flag = false;
        for (int i = 0; i < word.length(); i++)
        {
            if(word.charAt(i) == ch)
            {
                flag = true;
                pos = i;
                nextPos = pos + 1;
                break;
            }
        }
        if(flag){
            String updatedWord = "";
            for (int i = pos; i >= 0; i--)
                updatedWord+=word.charAt(i);
            if (pos != word.length() - 1)
            {
                for (int i = nextPos; i < word.length(); i++)
                    updatedWord+=word.charAt(i);
            }
            return updatedWord;
        }
        else return word;
    }
}